src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldSymbol.cs

```diff
@@ -390,5 +395,17 @@ private ConstantValue GetLazyConstantValue(bool earlyDecodingWellKnownAttributes
         }
 
         protected abstract ConstantValue MakeConstantValue(HashSet<SourceFieldSymbolWithSyntaxReference> dependencies, bool earlyDecodingWellKnownAttributes, BindingDiagnosticBag diagnostics);
+
+        internal override void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttributeData> boundAttributes, ImmutableArray<AttributeSyntax> allAttributeSyntaxNodes, BindingDiagnosticBag diagnostics, AttributeLocation symbolPart, WellKnownAttributeData decodedData)
+        {
+            if (this.state.HasComplete(CompletionPart.ConstantValue))
+            {
+                CheckConstantValue(_lazyConstantValue, diagnostics);
```

Comment:
>CheckConstantValue(_lazyConstantValue, diagnostics); [](http://example.com/codeflow?start=16&length=52)

It doesn't feel right that two different places are racing to perform this check. I think we should find an alternative place to perform it, the place that is not involved into a race.

Reply:
I need both attributes and constant value to be completed. I haven't found a place where this would be guaranteed.

Reply:
> I need both attributes and constant value to be completed.

I understand that. In ```SourceMemberFieldSymbol.ForceComplete```, the attributes are bound once we reach ```case CompletionPart.ConstantValue```. It feels like we can perform the check at the end of that case. But we want to make sure we do not produce duplicate diagnostics in a race condition. We might also take a look at how this is done for methods and follow the same approach

Reply:
> In `SourceMemberFieldSymbol.ForceComplete`, the attributes are bound once we reach `case CompletionPart.ConstantValue`. It feels like we can perform the check at the end of that case.

I haven't realized the `case`s are actually performed in order, thanks, that simplifies things. 

> We might also take a look at how this is done for methods and follow the same approach

Interesting, methods really emit some diagnostics in `ForceComplete`. But they perform locking on syntax which feels worse than my approach.

https://github.com/dotnet/roslyn/blob/0d6ace528d0a142ab51be350e05e651fd053b629/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs#L332-L359